the-collab-lab/tcl-37-smart-shopping-list
# Dp vd new item list

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description



**In this PR:**

- Viviana & I decided to merge the firebase codebase from the ListView component and make a single ListView component since both the groups made one separately. We did this to avoid merge conflicts in the future. 

- Then we re-factored the firebase codebase to instead go with a more React-style approach than a JavaScript approach as per Tanner's comments in Issue #1 PR. 

- We created a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon), and the data is stored as a token collection for a specific user.

- Then we incorporated some feedback for when the user submits the item, they are provided with a (imported react-toastify)feedback toast notification as per the wireframe example.

- All AC are passing with no issues

**To help us:** 
We used the example in https://github.com/CSFrequency/react-firebase-hooks/tree/v4.0.x/firestore to mimic the functionality in the #1 PR
## Related Issue



## Acceptance Criteria
AC:
* User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
* When the user submits the form, the item is saved to the database, associated with the user’s token
* Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
* The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)
* Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

![Screen Shot 2022-01-20 at 5 08 04 PM](https://user-images.githubusercontent.com/76231436/150430469-02457ce4-8f16-462a-8ef3-43b94921fc53.png)




### After
![Screen Shot 2022-01-20 at 5 08 13 PM](https://user-images.githubusercontent.com/76231436/150430482-484e0ef6-9838-41d0-b629-1b1c002f2dfc.png)
![Screen Shot 2022-01-20 at 5 11 03 PM](https://user-images.githubusercontent.com/76231436/150430486-ae4a551c-89d4-4539-b4de-017f92e0fbd1.png)



## Testing Steps / QA Criteria
- `npm install` 
- `npm start`
- When you fill out the form, you should expect an "Item added" notification
- When you check firestore, you should see a collection with the same name as your token
